app/models/booking_response.rb

```diff
@@ -40,7 +43,34 @@ def no_allowance?
     selection == 'no_allowance'
   end
 
+  def visitor_not_on_list?
+    selection == 'visitor_not_on_list'
+  end
+
+  def visitor_banned?
+    selection == 'visitor_banned'
+  end
+
   def slot_index
     SLOTS.index(selection)
   end
+
+  def unlisted_visitors
+    visitors.select { |v| unlisted_visitor_ids.include?(v.id) }
+  end
+
+  def banned_visitors
+    visitors.select { |v| banned_visitor_ids.include?(v.id) }
+  end
+
+private
+
+  def validate_checked_visitors
+    if selection == 'visitor_not_on_list' && unlisted_visitor_ids.empty?
+      errors.add :selection, :no_unlisted_visitors_selected
+    elsif selection == 'visitor_banned' && banned_visitor_ids.empty?
+      errors.add :selection, :no_banned_visitors_selected
+    end
+  end
+  validate :validate_checked_visitors
 end
```

Comment:
Good news: it turns out that once Virtus is told to use the correct attribute type, it will convert symbol inputs to strings. I've added a spec to prove this.